Remove GhostableViewController Conformances for Crash Investigation

soundsokay opened the pull request:
## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This attempts a fix to a top crash. See discussion on peaMlT-qR-p2. 

The root cause of the crash appears to be modifying the sublayers array of a CALayer while are iterating over it. An investigation revealed that the code causing the crash could be in `UIView+Ghost.swift` and `UIView+InnerGhost.swift`. Rather than attempting a fix directly which would require a fix in the WordPressUI pod, in this PR those types are used less often by removing conformances to `GhostableViewController`.

Of 13 total `GhostableViewController` conformances in the app 7 are removed here.

If the theory about the cause of the crash is correct the volume of "Collection CALayerArray: 0x2823ca370 was mutated while being enumerated crashes" should go down once this change is released.

Each `GhostableViewController` conformance removal was done on a view controller that also uses a `UIRefreshControl` for pull to refresh support. So while this change will remove the placeholder UI that `GhostableViewController` provides, the user has access to pull to refresh which if performed will reveal a loading indicator.

If this change can be merged and the crash goes down as predicted this PR can be reverted and a direct crash fix can be attempted later.

## Steps to reproduce
The crash so far has not been able to be reproduced.

## Testing information

Using the network link conditioner set to Very Bad Network allows this testing to be done more easily.

For any of the screens that no longer have the placeholder UI provided by `GhostableViewController`, tap into the screen and observe the placeholder is no longer shown.

During that time, swipe down on the screen to pull to refresh. Observe that a loading indicator appears.

A specific screen to try this on is the product variations screen: 

- tap into a variable product
- tap Variations
- observe no place holder UI is shown while the variations load, but that swiping down reveals a pull to refresh indicator.

To see the the placeholder UI try the product variations attributes screen which still conforms to `GhostableViewController`. 

- tap into a variable product
- tap Variations Attributes
- tap Add New Attribute
- observe that placeholder UI is shown briefly.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

For before and after images when adding attributes for a product variation see the comment "iOS CALayerArray Crash Investigation and Next Steps" on peaMlT-qR-p2


---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

soundsokay committed:
fix: remove GhostableViewController conformance

Fix attempt for crash: Collection <CALayerArray> was mutated while being enumerated

soundsokay committed:
Merge branch 'trunk' into dgalbraith/crash-fix-CALayerArray-mutated-while-enumerated

soundsokay committed:
Revert "fix: remove GhostableViewController conformance"

This reverts commit 21b48a568d61265bb72da83b6e9ede6d293af9ba.

soundsokay committed:
fix: remove GhostableViewController conformance from three product-related view controllers

This is part of a crash fix investigation for a top crash: Collection CALayerArray was mutated while being enumerated.

soundsokay committed:
fix: remove GhostableViewController conformance part II

This is part of a crash fix investigation for a top crash: Collection CALayerArray was mutated while being enumerated.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/210">19.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

soundsokay committed:
fix: Remove GhostableViewController conformance from coupon list and order list view controllers

This is part of a crash fix investigation for a top crash: Collection CALayerArray was mutated while being enumerated.

bozidarsevo reviewed (approved):
Just update the links (change to shortcodes)
